piquasso/_math/fock.py

```diff
@@ -46,17 +48,34 @@ def cutoff_cardinality(*, cutoff: int, d: int) -> int:
     return comb(d + cutoff - 1, cutoff - 1, exact=True)
 
 
-def cutoff_cardinality_array(*, cutoff, d):
-    return np.round(comb(d + cutoff - 1, d)).astype(int)
+@nb.njit
+def cutoff_cardinality_array(cutoff, d):
+    ret = np.empty(cutoff.shape, dtype=np.int32)
+
+    for i in range(len(cutoff)):
+        ret[i] = nb_comb(d + cutoff[i] - 1, d)
+
+    return ret
 
 
 def symmetric_subspace_cardinality_array(*, d: int, n: int) -> int:
     return np.round(comb(d + n - 1, n)).astype(int)
 
 
-@functools.lru_cache(maxsize=None)
```

Comment:
I think you don't have to delete this cache. The `cache=True` in `njit` is to avoid recompilation.

Reply:
Unfortunately, this function is called inside another `njit`-compiled function which results in an error since it cannot compile `@functools.lru_cache`:
`Cannot determine Numba type of <class 'functools._lru_cache_wrapper'>`

Reply:
One can reintroduce this by creating a different function which is cached. This function is called repeatedly during a calculation with mostly the same arguments. Even if it got faster (3-fold speedup in this case), it would be still nice to have it cached in cases when it is not used inside a jitted function.